### PR TITLE
checks added for :update_attribute for dynamic attributes refs #3875 …

### PR DESCRIPTION
**Pull request by Shwetakale**
…#3882

When we call process_attribute, it does all the checks for ensuring that dynamic attributes are allowed.

Note This impacts the :counter_cache option as it requires field defined on parent.

### CODE REVIEW

**Comment by johnnyshields** [comment]
@Shwetakale can you write a test case for this which uses an aliased field name, e.g.

```
field :foo, as: :my_field_name
```

and test that both `foo` and `my_field_name` can be set?

**Comment by Shwetakale** [comment]
@johnnyshields Sorry, my bad I missed this comment. I have updated pull request to check aliased field as well. Value of an aliased field is also set. Let me know if any other changes are required.

**Comment by johnnyshields** [comment]
@Shwetakale the PR is in an unmergable state. Can you please resolve conflicts?

**Comment by Shwetakale** [comment]
@johnnyshields Yes noticed that. I am resolving them.

**Comment by Shwetakale** [comment]
@johnnyshields Resolved conficts.

File: lib/mongoid/persistable/updatable.rb [hunk 1]
```diff
@@ -27,11 +27,7 @@ def update_attribute(name, value)
         as_writable_attribute!(name, value) do |access|
           normalized = name.to_s
           setter = "#{normalized}="
-          if respond_to?(setter)
-            send(setter, value)
-          else
-            write_attribute(access, value)
-          end
+          process_attribute(normalized, value)
```
Comment (johnnyshields) [inline lib/mongoid/persistable/updatable.rb]: The line `setter = "#{normalized}="` looks like its no longer needed here.

**Comment by johnnyshields** [comment]
@Shwetakale I don't see any tests for aliased fields in your latest PR.

**Comment by johnnyshields** [comment]
Ignore above comment I see it.

File: lib/mongoid/persistable/updatable.rb (hunk 1 above)
Comment (Shwetakale) [inline lib/mongoid/persistable/updatable.rb]: @johnnyshields Yes you are right. the line ```setter = "#{normalized}="``` is no longer needed here. My bad. I have removed that line and updated PR. Please check this now.

**Pull request merged**